File: Python/dicegame.py
import random

total = 0

while True:
    print('\nWelcome to the Dice Game!\n')
    num = int(input('Enter how many dice you want to roll: '))

    for x in range(num):
        rand = random.randint(1, 6)
        total += rand
        print('Die #' + str(x + 1) + ': ' + str(rand))

    print('\nTotal = ' + str(total))

    response = ''
    while response != 'Y' and response != 'N':
        print('\nWould you like to roll again? (Y/N)')
        response = input()
        if response == 'Y' or response == 'N':
            break
        print('Invalid response.')
    if response == 'N':
        break

print('Thanks for playing!')

File: Java/DiceGame.java
import java.util.Scanner;
import java.util.Random;

public class DiceGame {

  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    Random rand = new Random();
    int total = 0;

    System.out.println("Welcome to the Dice Game!\n");

    System.out.print("Enter how many dice you want to roll: ");
    int numDice = input.nextInt();
    System.out.println();

    for(int i = 0; i < numDice; i++) {
      int randomNumber = 1 + rand.nextInt(6);
      total += randomNumber;
      System.out.println("Die #" + ( i + 1 ) + ": " + randomNumber);
    }

    System.out.println("\nTotal = " + total);

  }

}
